packages/commerce-sdk-react/src/auth/storage/memory.ts

```diff
@@ -30,6 +30,10 @@ export class MemoryStorage extends BaseStorage {
     }
     delete(key: string) {
         const suffixedKey = this.getSuffixedKey(key)
-        this.map.delete(suffixedKey)
+        if (this.map.has(suffixedKey)) {
+            this.map.delete(suffixedKey)
+        } else {
+            console.error(`Failed to remove key, key not found: ${suffixedKey}`)
+        }
```

Comment:
```suggestion
        const wasPresent = this.map.delete(suffixedKey)
        if (wasPresent) console.warn(`Failed to remove key, key not found: ${suffixedKey}`)
```
[Map#delete](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete#return_value) returns the result of `.has` before the deletion occurred, so we don't need that check.

I'm not convinced, though, that we need to log anything here. It depends on your perspective on what "delete" does - does it guarantee that the state _changes_, or does it simply guarantee the end result (`key` is no longer present). In the former case, missing `key` is an error, because the state doesn't _change_. In the latter, though, missing `key` is fine, because we're already at the desired state.

Reply:
I removed the changes to the logs and left the changes to throwing errors and error handling for a separate PR.